articles/qos.md

```diff
@@ -0,0 +1,104 @@
+published: false
+author: '[Esteve Fernandez](https://github.com/esteve)'
+---
+
+* This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+With the advent of inexpensive robots using unreliable wireless networks, developers and users need mechanisms to control how traffic is prioritized across network links.
+
+## Background
+
+ROS 1.x uses TCP as the underlying transport, which is unsuitable for lossy networks such as wireless links.
+With ROS 2.0 relying on DDS which uses UDP as its transport, we can give control over the level of reliability a node can expect and act accordingly.
+
+## DDS Quality of Service policies
+
+DDS provides fine-grained control over the Quality of Service (QoS) setting for each of the entities involved in the system.
+Common entities whose behavior can be modified via QoS settings include: Topic, DataReader, DataWriter, Publisher and Subscriber.
+QoS is enforced based on a Request vs Offerer Model, however Publications and Subscriptions will only match if the QoS settings are compatible.
+
+## ROS 2.0 proposal
+
+Given the complexity of choosing the correct QoS settings for a given scenario, it may make sense for ROS 2.0 to provide a set of predefined QoS profiles for common usecases (e.g. sensor data, real time, etc.), while at the same time give the flexibility to control specific features of the QoS policies for the most common entities.
+
+## QoS profiles
+
+A QoS profile defines a set of policies, including durability, reliability, queue depth and sample history storage.
+The base QoS profile includes settings for the following policies:
+
+* History.
+  * Keep last: only store up to N samples, configurable via the queue depth option.
+  * Keep all: store all samples, subject to the configured resource limits of the DDS vendor.
+* Depth.
+  * Size of the queue: only honored if used together with "keep last".
+* Reliability.
+  * Best effort: attempt to deliver samples, but may lose them if the network is not robust.
+  * Reliable: guarantee that samples are delivered, may retry multiple times.
+* Durability.
+  * Transient local: *only applies to `DataWriter`s*. `DataWriter` becomes responsible of persisting samples until a corresponding `DataReader` becomes available.
+  * Volatile: no attempt is made to persist samples.
+
+Note: for each of the main bullets there is also the option of "system default", which uses whatever setting was defined via the DDS vendor tools (e.g. XML configuration files).
+
+ROS 2.0 will provide QoS profiles based on the following use cases:
+
+* Default QoS settings for publishers and subscriptions.
+In order to make the transition from ROS1 to ROS2, exercising a similar network behavior is desirable.
+By default, publishers and subscriptions are reliable in ROS2.
```

Comment:
Should we add note here that this default value won't work for larger messages for all rmw implementations atm?